documentation/local-dev/roslyn/readme.md

```diff
@@ -0,0 +1,37 @@
+# Instructions for Roslyn developers
+
+1. Check out a working copy of dotnet/roslyn, e.g. to the directory next to this repo's root `..\roslyn`.
+   Make sure the current working directory is the checkout of this repo.
+
+2. Copy LocalDev.Build.props to the root and set `RoslynRoot` property in the copied file to local Roslyn repo.
+
+    `cp documentation\local-dev\roslyn\LocalDev.Build.props LocalDev.Build.props`
+
+3. Copy LocalDev.Packages.props.roslyn to the root
+
+    `cp documentation\local-dev\roslyn\LocalDev.Packages.props LocalDev.Packages.props`
+
+4. Build Roslyn with:
+
+    `..\roslyn\Build.cmd -restore -pack`
```

Comment:
Consider a simpler approach (which is what I do currently and it works great):

1. Build Roslyn: `.\build.cmd -pack /p:VersionSuffix=$(git rev-parse --short HEAD)`
2. In SDK's Versions.props: find+replace (simple VSCode action) current roslyn version with the built version
3. Restore/build SDK.

That's it. No need to clean `.packages`. No need to keep the list of Version properties in LocalDev.Packages.props in sync with Versions.props (if we add more roslyn dependencies over time). Works in other repos too (like runtime or razor).

Reply:
It's not simpler. I don't need to build and pack the entire Roslyn from command line, which takes a long time. I can pack just one project from VS. I don't need to update version every time I make a change. I only need to delete Roslyn packages from .packages dir and restore. Ideally, I would not want to do that either.

Reply:
> I don't need to build and pack the entire Roslyn from command line, which takes a long time. I can pack just one project from VS.

I see, the doc says `build.cmd -pack` so I assumed that's what you are doing.

> I don't need to update version every time I make a change. I only need to delete Roslyn packages from .packages dir and restore.

Both are a single step, I don't think one is more difficult than the other.

But it's true that re-building just a single package with my approach isn't feasible. Your approach looks better in this scenario. Thanks for explaining.

Reply:
Another issue with using `$(git rev-parse --short HEAD)` is that you need to remember to commit each time you want to switch to the other repo.

Reply:
I think setting VersionSuffix to current time-based value would make it simpler still. Might explore that later on.